txsort.go

```diff
@@ -0,0 +1,100 @@
+// TxSort
+// Provides functions for sorting tx inputs and outputs according to BIP LI01
+
+package btcutil
+
+import (
+	"bytes"
+	"sort"
+
+	"github.com/btcsuite/btcd/wire"
+)
+
+// TxSort sorts the inputs and outputs of a tx based on BIP LI01
+// It does not modify the transaction given, but returns a new copy
+// which has been sorted and may have a different txid.
+func TxSort(tx *wire.MsgTx) *wire.MsgTx {
+	txCopy := tx.Copy()
+	sortInputs(txCopy.TxIn)
```

Comment:
I'd suggest just doing `sort.Sort(sortableInputSlice(txCopy.TxIn))` here.  In my experience, the Go compiler is not particularly good about inlining.  Also, I noticed the called function is returning a value that isn't used.